### PR TITLE
Add release announcement for Flink CDC 3.1.0

### PR DESCRIPTION
**Pull request by PatrickRen**
This pull request adds release announcement blog post for Flink CDC 3.1.0.

### CODE REVIEW

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+## Highlights
+
+### Transformation Support in Pipeline
+
+Flink CDC 3.1.0 introduces the ability of making transformations in the CDC pipeline. By incorporating a `transform` section within the YAML pipeline definitions, users can now easily apply a variety of transformations to data change event from source, including projections, calculations, and addition of constant columns, enhancing the effectiveness of data intergration pipelines. Leveraging an SQL-like syntax for defining these transformations, the new feature ensures that users can quickly adapt to and utilize it.
```
Comment (morazow) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]:
```suggestion
Flink CDC 3.1.0 introduces the ability of making transformations in the CDC pipeline. By incorporating a `transform` section within the YAML pipeline definitions, users can now easily apply a variety of transformations to data change event from source, including projections, calculations, and addition of constant columns, enhancing the effectiveness of data integration pipelines. Leveraging an SQL-like syntax for defining these transformations, the new feature ensures that users can quickly adapt to and utilize it.
```

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+### Table Merging Support
+
+Flink CDC 3.1.0 now suuports merging multiple tables into one by configuring `route` in the YAML pipeline definition.  It is a prevalent occurrence where business data is partitioned across tables even databases due to the substantial volume. By configuring `route`s that mapping multiple tables into one, data change events will be merged into the same destination table. Moreover, schema changes on source tables will also be applied to the destination.
```
Comment (morazow) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]:
```suggestion
Flink CDC 3.1.0 now supports merging multiple tables into one by configuring `route` in the YAML pipeline definition.  It is a prevalent occurrence where business data is partitioned across tables even databases due to the substantial volume. By configuring `route`s that map multiple tables into one, data change events will be merged into the same destination table. Moreover, schema changes on source tables will also be applied to the destination.
```

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#### Distributions of MySQL / Oracle / OceanBase / Db2 connectors
+
+Unfortunately due to the license incompatibility, we cannot ship JDBC drivers of the following connectors together with our binary release:
+
+- Db2
+- MySQL
+- Oracle
+- OceanBase
+
+Please manually download the corresponding JDBC driver into `$FLINK_CDC_HOME/lib` and `$FLINK_HOME/lib`, and specify their paths when submiting YAML pipelines with `--jar`, or make sure they are under the classpath if you are using Flink SQL.
```
Comment (morazow) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]:
```suggestion
Please manually download the corresponding JDBC driver into `$FLINK_CDC_HOME/lib` and `$FLINK_HOME/lib`, and specify their paths when submitting YAML pipelines with `--jar`, or make sure they are under the classpath if you are using Flink SQL.
```

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+#### SinkFunction Support
+
+Although `SinkFunction` has been marked as deprecated in Flink, considering some connectors are still using the API, we also supports `SinkFunction` API for CDC pipeline sinks to help expanding the ecosystem of Flink CDC.
```
Comment (morazow) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]:
```suggestion
Although `SinkFunction` has been marked as deprecated in Flink, considering some connectors are still using the API, we also support `SinkFunction` API for CDC pipeline sinks to help expand the ecosystem of Flink CDC.
```

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+#### New Pipeline Connectors
+
+Flink CDC 3.1.0 introduces 2 new pipeline connectors:
+
+- Apache Kafka sink
+- Apache Paimon sink
+
+#### MySQL
+
+In this release, MySQL pipeline source introduces a new option `tables.exclude` to exclude unnecessary tables from capturing with an easier expression. MySQL CDC sources is now shipped with a custom converter `MysqlDebeziumTimeConverter` for converting temporal type columns to a more human-readable and serialize-friendly string.
```
Comment (morazow) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]:
```suggestion
In this release, MySQL pipeline source introduces a new option `tables.exclude` to exclude unnecessary tables from capturing with an easier expression. MySQL CDC source is now shipped with a custom converter `MysqlDebeziumTimeConverter` for converting temporal type columns to a more human-readable and serialize-friendly string.
```

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+OceanBase CDC source now supports specifying the general `DebeziumDeserializationSchema` for reusing existing Debezium deserializers.
+
+#### Db2
+
+Db2 CDC source is now migrated to the unified incremental snapshot framework.
+
+### CLI
+
+Flink CDC pipeline submission CLI now supports recovering a pipeline exection from a specific savepoint file by using command line argument `--from-savepoint`
```
Comment (morazow) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]:
```suggestion
Flink CDC pipeline submission CLI now supports recovering a pipeline execution from a specific savepoint file by using command line argument `--from-savepoint`
```

**Review by morazow — CHANGES_REQUESTED** [review]
@PatrickRen thanks for the release preparation 🥳 

I have added minor suggestions

**Review by Jiabao-Sun — APPROVED** [review]

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+- renqs:
+  name: "Qingsheng Ren"
+  twitter: "renqstuite"
+aliases:
+- /news/2024/MM/DD/release-cdc-3.1.0.html
+---
+
+The Apache Flink community is excited to announce the release of Flink CDC 3.1.0! This is the first release after the community accepted the donation of Flink CDC as a sub-project of Apache Flink, with exciting new features such as transform and table merging. The eco-system of Flink CDC keeps expanding, including new Kafka and Paimon pipeline sinks and enhancement to existing connectors.
+
+We'd like to invite you to check out [Flink CDC documentation](https://nightlies.apache.org/flink/flink-cdc-docs-stable) and have a try on [the quickstart tutorial](https://nightlies.apache.org/flink/flink-cdc-docs-release-3.0/docs/get-started/introduction) to explore the world of Flink CDC. Also we encourage you to [download the release](https://flink.apache.org/downloads.html) and share your feedback with the community through the Flink [mailing lists](https://flink.apache.org/community.html#mailing-lists) or [JIRA](https://issues.apache.org/jira/browse/flink)! We hope you like the new release and we’d be eager to learn about your experience with it.
```
Comment (JingsongLi) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: link release-3.1 document?

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 1 above)
Comment (JingsongLi) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: Do you think it is better to add an example?

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 2 above)
Comment (JingsongLi) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: Ditto, do you think it is better to add an example?

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+#### SinkFunction Support
+
+Although `SinkFunction` has been marked as deprecated in Flink, considering some connectors are still using the API, we also supports `SinkFunction` API for CDC pipeline sinks to help expanding the ecosystem of Flink CDC.
+
+#### New Pipeline Connectors
+
+Flink CDC 3.1.0 introduces 2 new pipeline connectors:
+
+- Apache Kafka sink
+- Apache Paimon sink
```
Comment (JingsongLi) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: maybe note paimon version here?

**Comment by JingsongLi** [comment]
Thanks @PatrickRen , Left some comments.

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 7 above)
Comment (leonardBang) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]:
```suggestion
We'd like to invite you to check out [Flink CDC documentation](https://nightlies.apache.org/flink/flink-cdc-docs-stable) and have a try on [the quickstart tutorial](https://nightlies.apache.org/flink/flink-cdc-docs-stable/docs/get-started/introduction) to explore the world of Flink CDC. Also we encourage you to [download the release](https://flink.apache.org/downloads.html) and share your feedback with the community through the Flink [mailing lists](https://flink.apache.org/community.html#mailing-lists) or [JIRA](https://issues.apache.org/jira/browse/flink)! We hope you like the new release and we’d be eager to learn about your experience with it.
```

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 8 above)
Comment (leonardBang) [reply docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: +1

**Review by leonardBang — COMMENTED** [review]
Thanks @PatrickRen for the contribution, I left two minor comments

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 3 above)
Comment (wuchong) [reply docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: It sounds like we have to put driver in `$FLINK_CDC_HOME/lib` and `$FLINK_HOME/lib` and submit YAML with `--jar`, to make it work. Does it work with any of them?

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 7 above)
Comment (wuchong) [inline docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]:
```suggestion
We'd like to invite you to check out [Flink CDC documentation](https://nightlies.apache.org/flink/flink-cdc-docs-stable) and have a try on [the quickstart tutorial](https://nightlies.apache.org/flink/flink-cdc-docs-release-3.0/docs/get-started/introduction) to explore the world of Flink CDC. Also we encourage you to [download the release](https://flink.apache.org/downloads.html#flink-cdc) and share your feedback with the community through the Flink [mailing lists](https://flink.apache.org/community.html#mailing-lists) or [JIRA](https://issues.apache.org/jira/browse/flink)! We hope you like the new release and we’d be eager to learn about your experience with it.
```

Add a `#flink-cdc` anchor to the download URL to help user locate the download items easier.

**Review by wuchong — COMMENTED** [review]
Should we need to update the Download page to include the flink CDC release binaries?

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 3 above)
Comment (PatrickRen) [reply docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: Yep I will rewrite this part. There are two ways to involve MySQL JDBC driver:

1. Put it into `FLINK_HOME/lib` directly
2. Download to anywhere locally and specify the path with `--jar`. It is not necessarily in `FLINK_CDC_HOME/lib`

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 7 above)
Comment (PatrickRen) [reply docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: Sure! I'll create another PR for download links

File: docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md (hunk 1 above)
Comment (PatrickRen) [reply docs/content/posts/2024-MM-DD-release-cdc-3.1.0.md]: Thanks for the suggestion! I added a link to examples in the documentation.

**Pull request merged**